llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/metrics/LlapTaskSchedulerMetrics.java

```diff
@@ -276,6 +302,43 @@ private void getTaskSchedulerStats(MetricsRecordBuilder rb) {
         .addCounter(SchedulerPendingPreemptionTaskCount, pendingPreemptionTasksCount.value())
         .addCounter(SchedulerPreemptedTaskCount, preemptedTasksCount.value())
         .addCounter(SchedulerCompletedDagCount, completedDagcount.value());
+    daemonTaskLatency.forEach((k, v) -> rb.addGauge(v, v.getMean()));
+  }
+
+  static class DaemonLatencyMetric implements MetricsInfo {
+    private String name;
+    private ExponentiallyDecayingReservoir reservoir;
```

Comment:
What is the benefit of using am exponential decay here, vs. simple sliding window (i.e. guava stats)? Specifically with the low amount of entries (right now 1028), a few more recent task execution times can have a huge impact on what we believe is the node's average over a longer period of time. The whole idea of using the average here is to smooth out the task execution times across a larger amount of tasks. I have the fear, that the exponential decay is contradicting this by giving few, very recent values, a high impact.

Reply:
Do not like Guava, and Stats appeared in guava 20. We are on 19 now and upgrading can cause problems with other projects like hadoop.
Moved to using apache math DescriptiveStatistics instead.